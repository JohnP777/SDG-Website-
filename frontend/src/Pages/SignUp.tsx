import SignUpForm from '../Components/SignUpForm';
import { Box } from '@mui/system';
import Page from '../Components/Page';
import { Button, Container, Typography } from '@mui/material';
import { useNavigate } from 'react-router-dom';

const SignUp = () => {

  const navigate = useNavigate();

  return (
    <Page center>
      <Container
        maxWidth='md'
        disableGutters
        sx={{
          px: { xs: 2, sm: 3 }
        }}
      >
        <Box sx={{
          display: 'flex',
          flexDirection: { xs: 'column', sm: 'row' },
          flexGrow: 1,
          justifyContent: 'center',
          alignItems: 'stretch',
          width: '100%',
          maxWidth: '100%',
        }}>
          {/* Blue Box on left (that lets you navigate to login page) */}
          <Box sx={{
            background: '#4285F4',
            width: { xs: '100%', sm: '35%' },
            maxWidth: 400,
            boxSizing: 'border-box',
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
            justifyContent: 'center',
            padding: 3,
            gap: 2,
            borderRadius: { xs: '8px 8px 0 0', sm: '8px 0 0 8px' }
          }}>
            <Typography variant='h4' sx={{ fontWeight: 'bold', color: 'white', textAlign: 'center' }}>Hello Friend!</Typography>
            <Typography variant='h5' sx={{ fontWeight: 'bold', color: 'white', textAlign: 'center' }}>Already a user?</Typography>
            <Button
              variant='contained'
              size='small' 
              sx={{
                backgroundColor: 'white',
                width: '85%',
                fontSize: '10px',
                textTransform: 'none',
                boxShadow: 'none',
                color: '#4285F4'
              }}
              onClick={() => navigate('/login')}
            >
              Login
            </Button>
          </Box>
          <SignUpForm />
        </Box>
      </Container>
    </Page>
  );
};

export default SignUp;