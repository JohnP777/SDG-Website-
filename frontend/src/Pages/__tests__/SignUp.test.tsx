import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import SignUp from "../SignUp";
import { MemoryRouter } from "react-router-dom";
import { GoogleOAuthProvider } from '@react-oauth/google';
  
// Mock window.location.reload
const mockReload = jest.fn();
Object.defineProperty(window, 'location', {
  value: {
    reload: mockReload,
  },
  writable: true
});

// Mock useNavigate
const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

// Helper function to render the component with router
const renderWithRouter = (component: React.ReactNode) => {
return render(
    <GoogleOAuthProvider clientId="google-client-id">
      <MemoryRouter>
        {component}
      </MemoryRouter>
    </GoogleOAuthProvider>
);
};

describe("SignUp", () => {

  // Snapshot test
  it("should match snapshot", async () => {
    const { asFragment } = renderWithRouter(<SignUp />);
    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });

  // Check if SignUp text is on screen
  it("signup text should be on screen", () => {
      renderWithRouter(<SignUp />);
      
      expect(screen.getByText("Hello Friend!")).toBeInTheDocument();
      expect(screen.getByText("Already a user?")).toBeInTheDocument();
      expect(screen.getByText("Login")).toBeInTheDocument();
  
  });

  it("should navigate to login page when login up button is clicked", () => {
    renderWithRouter(<SignUp />);
    
    const signUpButton = screen.getByText("Login");
    fireEvent.click(signUpButton);
    
    expect(mockedUsedNavigate).toHaveBeenCalledWith("/login");

  });
});