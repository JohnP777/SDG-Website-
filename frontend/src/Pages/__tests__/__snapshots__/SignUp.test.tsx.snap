// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-z3339j"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-u2vtxm-MuiPaper-root-MuiAppBar-root"
      style="--Paper-shadow: none;"
    >
      <div
        class="MuiContainer-root css-12pwiy3-MuiContainer-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular css-1t0x8et-MuiToolbar-root"
        >
          <a
            class="MuiTypography-root MuiTypography-subtitle1 css-1z0p208-MuiTypography-root"
            href="/"
          >
            SDG
            <br />
            ZOO
          </a>
          <div
            class="MuiBox-root css-dq25au"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-z77o6z-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1xf0eoi-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                />
              </svg>
            </button>
          </div>
          <div
            class="MuiBox-root css-10s323d"
          >
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1oyog3l-MuiButtonBase-root-MuiButton-root"
              href="/"
              tabindex="0"
            >
              HOME
            </a>
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1oyog3l-MuiButtonBase-root-MuiButton-root"
              href="/about-us"
              tabindex="0"
            >
              ABOUT
            </a>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-769nkz-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              FEATURES
              <span
                class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-1gulhci-MuiButton-endIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="KeyboardArrowDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="MuiBox-root css-1t0e2o"
          >
            <button
              aria-label="Profile settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1avsmgh-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-bqh3fr-MuiAvatar-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="PersonIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4m0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4"
                  />
                </svg>
              </div>
            </button>
          </div>
        </div>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-g3sf3s-MuiContainer-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthMd MuiContainer-disableGutters css-1lgrfpo-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-quy04j"
        >
          <div
            class="MuiBox-root css-18p0lj7"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-1lbur98-MuiTypography-root"
            >
              Hello Friend!
            </h4>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-evdd39-MuiTypography-root"
            >
              Already a user?
            </h5>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-1yrvhw5-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Login
            </button>
          </div>
          <div
            class="MuiBox-root css-1gaussg"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
              style="font-size: 1.6rem; font-weight: bold; margin-bottom: 20px; text-align: center;"
            >
              SDG Knowledge Sign Up
            </p>
            <div
              class="MuiBox-root css-1x1nri4"
            >
              <input
                name="username"
                placeholder="Username"
                style="width: 85%; padding: 5px 10px; font-size: 12px; border: 1px solid #e0e0e0; border-radius: 5px; box-sizing: border-box;"
                type="text"
                value=""
              />
              <input
                name="contact"
                placeholder="Email"
                style="width: 85%; padding: 5px 10px; font-size: 12px; border: 1px solid #e0e0e0; border-radius: 5px; box-sizing: border-box;"
                type="text"
                value=""
              />
              <input
                name="phonenumber"
                placeholder="Phone Number (optional)"
                style="width: 85%; padding: 5px 10px; font-size: 12px; border: 1px solid #e0e0e0; border-radius: 5px; box-sizing: border-box;"
                type="text"
                value=""
              />
              <input
                name="password"
                placeholder="Password"
                style="width: 85%; padding: 5px 10px; font-size: 12px; border: 1px solid #e0e0e0; border-radius: 5px; box-sizing: border-box;"
                type="password"
                value=""
              />
              <input
                name="confirmpassword"
                placeholder="Confirm Password"
                style="width: 85%; padding: 5px 10px; font-size: 12px; border: 1px solid #e0e0e0; border-radius: 5px; box-sizing: border-box;"
                type="password"
                value=""
              />
              <label
                class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1ejnzjp-MuiFormControlLabel-root"
              >
                <span
                  class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
                >
                  <input
                    class="PrivateSwitchBase-input css-j8yymo"
                    data-indeterminate="false"
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                >
                  Show Password
                </span>
              </label>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-1eioe9r-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Sign Up
              </button>
              <div
                aria-orientation="horizontal"
                class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-jyffef-MuiDivider-root"
                role="separator"
              >
                <span
                  class="MuiDivider-wrapper css-1134oje-MuiDivider-wrapper"
                >
                  or continue with
                </span>
              </div>
              <div
                style="height: 32px;"
              />
              <div
                style="color: rgb(130, 130, 130); font-size: 10px; margin-top: 5px; text-align: center;"
              >
                By clicking continue, you agree to our 
                <a
                  href="https://policies.google.com/terms?hl=en"
                  style="color: rgb(0, 0, 0);"
                >
                  Terms of Service
                </a>
                  and 
                <a
                  href="https://policies.google.com/privacy?hl=en"
                  style="color: rgb(0, 0, 0);"
                >
                  Privacy Policy
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
